### PR TITLE
Automatically fill padding with poison when loading an aggregate value

### PR DESCRIPTION
**Pull request by aqjune**
No regression from LLVM unit tests.

I'll run single file benchmarks too & see whether there is any failure.
If there is no failure, I think I can send the suggestion to llvm-dev.

### CODE REVIEW

**Comment by aqjune** [comment]
Has no regression from single file benchmark (finds a few bugs which are known to be unsound transformations). I think this patch is fine.

When talking about this with LLVM people, I'd like to say that the loaded padding should be undef first. The reason is that saying padding should be poison seems to be an orthogonal issue from this.

File: ir/memory.cpp [hunk 1]
```diff
@@ -1947,13 +1947,19 @@ StateValue Memory::load(const Pointer &ptr, const Type &type, set<expr> &undef,
...
+      }
+
       auto ptr_i = ptr + byteofs;
       auto align_i = gcd(align, byteofs % align);
       member_vals.emplace_back(load(ptr_i, aty->getChild(i), undef, align_i));
       byteofs += getStoreByteSize(aty->getChild(i));
     }
     assert(byteofs == bytecount);
-    return aty->aggregateVals(member_vals);
+    return aty->aggregateVals(member_vals, true);
```
Comment (nunoplopes) [inline ir/memory.cpp:1962]: is there any use of aggregateVals(foo, false) left?

Comment (aqjune) [reply ir/memory.cpp:1962]: AggregateValue::toSMT and AggregateType::fromInt use it; In case of other places it seems giving false is not necessary.

**Comment by aqjune** [comment]
The LangRef update landed: https://reviews.llvm.org/rG98e5776897ce

**Pull request merged**